server/src/storage/mod.rs

```diff
@@ -403,4 +506,110 @@ mod tests {
+        let temp_dir = tempdir().unwrap();
+        let batch = test_util::get_compressed_segment_record_batch();
+
+        let parquet_path = temp_dir.path().join("test.parquet");
+        StorageEngine::write_batch_to_apache_parquet_file(batch, parquet_path.as_path());
+
+        assert!(parquet_path.exists());
+    }
+
+    #[test]
+    fn test_write_empty_batch_to_apache_parquet_file() {
+        let schema = Schema::new(vec![]);
+        let batch = RecordBatch::new_empty(Arc::new(schema));
+
+        let temp_dir = tempdir().unwrap();
+        let parquet_path = temp_dir.path().join("empty.parquet");
+        StorageEngine::write_batch_to_apache_parquet_file(batch, parquet_path.as_path());
+
+        assert!(parquet_path.exists());
+    }
+
+    #[test]
+    fn test_write_batch_to_file_with_invalid_extension() {
+        write_to_file_and_assert_failed("test.txt".to_owned());
+    }
+
+    #[test]
+    fn test_write_batch_to_file_with_no_extension() {
+        write_to_file_and_assert_failed("test".to_owned());
+    }
+
+    fn write_to_file_and_assert_failed(file_name: String) {
+        let temp_dir = tempdir().unwrap();
+        let batch = test_util::get_compressed_segment_record_batch();
+
+        let parquet_path = temp_dir.path().join(file_name);
+        let result =
+            StorageEngine::write_batch_to_apache_parquet_file(batch, parquet_path.as_path());
+
+        assert!(result.is_err());
+        assert!(!parquet_path.exists());
+    }
+
+    #[test]
+    fn test_read_batch_from_apache_parquet_file() {
+        let file_name = "test.parquet".to_owned();
+        let (_temp_dir, path, batch) = create_file_in_temp_dir(file_name);
+
+        let result = StorageEngine::read_batch_from_apache_parquet_file(path.as_path());
+
+        assert!(result.is_ok());
+        assert_eq!(batch, result.unwrap());
+    }
+
+    #[test]
+    fn test_read_batch_from_non_parquet_file() {
+        let temp_dir = tempdir().unwrap();
+        let path = temp_dir.path().join("test.txt");
+        File::create(path.clone()).unwrap();
+
+        let result = StorageEngine::read_batch_from_apache_parquet_file(path.as_path());
+
+        assert!(result.is_err());
+    }
+
+    #[test]
+    fn test_read_batch_from_non_existent_path() {
+        let temp_dir = tempdir().unwrap();
+        let path = temp_dir.path().join("none.parquet");
+        let result = StorageEngine::read_batch_from_apache_parquet_file(path.as_path());
+
+        assert!(result.is_err());
+    }
+
+    #[test]
+    fn test_is_parquet_path_apache_parquet_file() {
+        let file_name = "test.parquet".to_owned();
+        let (_temp_dir, path, _batch) = create_file_in_temp_dir(file_name);
+
+        assert!(StorageEngine::is_path_an_apache_parquet_file(path.as_path()));
+    }
+
+    /// Create an Apache Parquet file from a generated record batch in the temp dir.
+    fn create_file_in_temp_dir(file_name: String) -> (TempDir, PathBuf, RecordBatch) {
```

Comment:
Maybe rename to `create_apache_parquet_file_in_temp_dir()`.

Reply:
Fixed.